src/depends/safeserver/safehttpserver.cpp

```diff
@@ -9,35 +9,44 @@
 
 #include "safehttpserver.h"
 #include <cstdlib>
-#include <sstream>
 #include <cstring>
 #include <iostream>
+#include <sstream>
+#include "common/Constants.h"
 #include "jsonrpccpp/common/specificationparser.h"
 #include "libUtils/Logger.h"
-#include "common/Constants.h"
 
 using namespace jsonrpc;
 using namespace std;
 
-#define BUFFERSIZE 65536
+// too many connections may occur and thread poll sizes can be limited
+static const int LISTEN_BACKLOG_SIZE = 256;
 
 struct mhd_coninfo {
-    struct MHD_PostProcessor *postprocessor;
-    MHD_Connection* connection;
-    stringstream request;
-    SafeHttpServer* server;
-    int code;
+  struct MHD_PostProcessor *postprocessor;
+  MHD_Connection *connection;
+  stringstream request;
+  SafeHttpServer *server;
+  int code;
 };
 
-SafeHttpServer::SafeHttpServer(int port, const std::string &sslcert, const std::string &sslkey, int threads) :
-    AbstractServerConnector(),
-    port(port),
-    threads(threads),
-    running(false),
-    path_sslcert(sslcert),
-    path_sslkey(sslkey),
-    daemon(NULL),
-    bindlocalhost(false) {}
+SafeHttpServer::SafeHttpServer(int port, int threads,
+                               const std::string &sslcert,
+                               const std::string &sslkey)
+    : AbstractServerConnector(),
+      port(port),
+      threads(threads),
+      running(false),
+      path_sslcert(sslcert),
+      path_sslkey(sslkey),
+      daemon(NULL),
+      bindlocalhost(false) {
+  if (threads < 1) {
+    this->threads = 1;
+  } else if (threads > 50) {
+    this->threads = 50;
```

Comment:
Why are we capping threads at 50 here? How do we know this won't cause problems in prod if all threads get busy?